packages/workspace/src/common/test/mock-workspace-server.ts

```diff
@@ -19,7 +19,15 @@ import { WorkspaceServer } from '../workspace-protocol';
 @injectable()
 export class MockWorkspaceServer implements WorkspaceServer {
 
-    getRoot(): Promise<string | undefined> { return Promise.resolve(''); }
+    getWorkspace(): Promise<string | undefined> { return Promise.resolve(''); }
 
-    setRoot(uri: string): Promise<void> { return Promise.resolve(); }
+    setWorkspace(uri: string): Promise<void> { return Promise.resolve(); }
+
+    addRoot(rootUriToAdd: string, workspaceFolderUri: string): Promise<void> { return Promise.resolve(); }
```

Comment:
An idea was to remove `addRoot/removeRoot/getRoots/getWorkspaceConfigFilePath` and do it in the frontend only,  in `WorkspaceService`. So if `getWorkspace` points to a workspace config file, then we should read and modify it via `FileSystem`, and listen to changes via `FileSystemWatcher`.

Reply:
While I was doing refactoring I had kept thinking what's the boundary between the FE and BE in this project.

What I wanted to do initially is letting FE call the BE when it needs something, and let the BE to do that job. After the refactoring, the BE, however, has no knowledge of workspace folders being opened, or the roots added. 

To be honest, I have no preference on one way of doing things than the other. But this makes me think: If we count on the `workspace-service` in FE to do the job, why do we need `default-workspace-server` or `workspace-protocol` (in BE) at all? Reading and writing content from / to `recentworkspaces.json` can be done in FE as well, by calling `FileSystem` from the FE. 

Could you please share your insights on this? @akosyakov

Reply:
please file an issue to discuss it that it does not get lost